x-pack/plugins/apm/server/routes/alerts/rule_types/anomaly/get_anomaly_detector_type_index.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ApmMlDetectorType,
+  getApmMlDetectorIndex,
+} from '../../../../../common/anomaly_detection/apm_ml_detectors';
+import { AnomalyAlertDetectorType } from '../../../../../common/rules/apm_rule_types';
+
+export function getAnomalyDetectorTypeIndex(
+  detectorTypes: AnomalyAlertDetectorType[]
+) {
+  return detectorTypes.map((detector) => {
+    switch (detector) {
+      case 'latency':
+        return getApmMlDetectorIndex(ApmMlDetectorType.txLatency);
+      case 'throughput':
+        return getApmMlDetectorIndex(ApmMlDetectorType.txThroughput);
+      case 'failed_transaction_rate':
+        return getApmMlDetectorIndex(ApmMlDetectorType.txFailureRate);
+    }
+  });
+}
```

Comment:
I don't get why this is needed. Can't we use `getApmMlDetectorIndex` directly ?

Reply:
I don't see where this is used, so it's a little for me to guess why it's needed

Reply:
There's a difference between the params I created for the rules and this ML object
```
export enum ApmMlDetectorType {
  txLatency = 'txLatency',
  txThroughput = 'txThroughput',
  txFailureRate = 'txFailureRate',
}
```
I can try to reuse this object as key, and see how I can render the options and keep the values without using the `type` but the `label`